### DISCUSSION TITLE
Connecting Statamic to our custom api

### DISCUSSION THREAD

**Discussion by justageek** [statamic/cms#6020] [General]
We are building a custom e-commerce / shopping platform and it will provide a headless experience through a REST api.  I'm wondering how viable it might be to use Statamic to provide custom front end sites consuming our API to our customers, rather than trying to build our own UI and theming layer.  Would appreciate any thoughts on this idea.

Thanks,
Brian

**Comment by edalzell** [comment]
Not quite sure what you mean by "UI and theming layer". Statamic does have both a REST and GraphQL API, but then you're in headless mode so end users would be in charge of their own UI?

What UI are you talking about, the CP?

**Comment by ryanmitchell** [comment]
If I'm understanding you right I think you mean you want Statamic to provide the CMS functions for your users, and include the data from your API for them to select within the UI (eg for products, orders etc). If I'm correct, then yes that is possible to do - you have the ability to [make custom FieldTypes](https://statamic.dev/extending/fieldtypes) giving the ability to include data from your API. You could even leverage work done on extensions like [Runway](https://github.com/doublethreedigital/runway) which pulls in data from a database - you would change that to be your API.

**Reply by justageek** [reply]
Ok, thanks, what I'm mainly trying to achieve is a single code base / application that serves a multi-tenant application and allows each user / client to choose a theme for their shopping cart, similar to how a Shopify would work.  I know I can build that with a more traditional Laravel web application, was just wondering if Statamic might already include this "choose your theme" concept out of the box, which I could then wire up in headless mode to our e-commerce api.
